src/components/side_panel/pivot/pivot_dimensions/add_dimension_button/add_dimension_button.ts

```diff
@@ -65,20 +104,41 @@ export class AddDimensionButton extends Component<Props, SpreadsheetChildEnv> {
     };
   }
 
+  updateSearch(searchInput: string) {
+    this.search.input = searchInput;
+    this.autoComplete.useProvider(this.getProvider());
+  }
+
   pickField(field: PivotField) {
     this.props.onFieldPicked(field.name);
-    this.popover.isOpen = false;
-    this.search.input = "";
+    this.togglePopover();
   }
 
   togglePopover() {
     this.popover.isOpen = !this.popover.isOpen;
     this.search.input = "";
+    this.autoComplete.useProvider(this.getProvider());
   }
 
   onKeyDown(ev: KeyboardEvent) {
-    if (this.filteredFields.length === 1 && ev.key === "Enter") {
-      this.pickField(this.filteredFields[0]);
+    switch (ev.key) {
+      case "Enter":
+        const proposals = this.autoComplete.provider?.proposals;
+        if (proposals?.length === 1) {
+          this.autoComplete.provider?.selectProposal(proposals[0].text || "");
+        }
+        const proposal = this.autoComplete.selectedProposal;
+        this.autoComplete.provider?.selectProposal(proposal?.text || "");
+        break;
+      case "ArrowUp":
+      case "ArrowDown":
+        this.autoComplete.moveSelection(ev.key);
```

Comment:
```suggestion
        this.autoComplete.moveSelection(ev.key === "ArrowDown" ? "next" : "previous");
```

Reply:
Oups, fixed :sweat_smile: 
The test is on odoo's side until pivots lands in o-spreadsheet
https://github.com/odoo/enterprise/pull/61671/files#diff-4b8b6f5d4dffedcb22e750296d4e58fb62e198e42789e82d1f1d6a28d4bdbc9dR511